fenix/app/src/test/java/org/mozilla/fenix/components/toolbar/EngineViewClippingBehaviorTest.kt

```diff
@@ -0,0 +1,284 @@
+package org.mozilla.fenix.components.toolbar
+
+import android.view.View
+import android.widget.EditText
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.browser.toolbar.BrowserToolbar
+import mozilla.components.concept.engine.EngineView
+import mozilla.components.support.test.fakes.engine.FakeEngineView
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.mozilla.fenix.components.toolbar.navbar.EngineViewClippingBehavior
+import org.mozilla.fenix.components.toolbar.navbar.ToolbarContainerView
+
+@RunWith(AndroidJUnit4::class)
+class EngineViewClippingBehaviorTest {
+
+    // Bottom toolbar position tests
+    @Test
+    fun `GIVEN the toolbar is at the bottom WHEN toolbar is being shifted THEN EngineView adjusts bottom clipping && EngineViewParent position doesn't change`() {
+        val engineView: EngineView = Mockito.spy(FakeEngineView(testContext))
+        val engineParentView: View = Mockito.spy(View(testContext))
+        val toolbar: BrowserToolbar = mock()
+        Mockito.doReturn(Y_DOWN_TRANSITION).`when`(toolbar).translationY
+
+        Assert.assertEquals(0f, engineParentView.translationY)
```

Comment:
```suggestion
       doReturn(Y_DOWN_TRANSITION).`when`(toolbar).translationY

       assertEquals(0f, engineParentView.translationY)
```
Could you import the subclasses for these instead of relying on the top-level `Mockito`